lib/cloud/azure/vm.go

```diff
@@ -166,3 +167,74 @@ func (c *vmClient) ListVirtualMachines(ctx context.Context, resourceGroup string
 
 	return virtualMachines, nil
 }
+
+// vmRunCommand provides an interface for an Azure Run Command client.
+type vmRunCommand interface {
+	BeginCreateOrUpdate(ctx context.Context, resourceGroup, vmName, runCommandName string, runCommand armcompute.VirtualMachineRunCommand, options *armcompute.VirtualMachineRunCommandsClientBeginCreateOrUpdateOptions) (*runtime.Poller[armcompute.VirtualMachineRunCommandsClientCreateOrUpdateResponse], error)
+}
+
+// RunCommandRequest combines parameters for running a command on an Azure virtual machine.
+type RunCommandRequest struct {
+	// Region is the region of the VM.
+	Region string
+	// ResourceGroup is the resource group for the VM.
+	ResourceGroup string
+	// VMName is the name of the VM.
+	VMName string
+	// Script is the URI of the script for the virtual machine to execute.
+	Script string
+	// Parameters is a list of parameters for the script.
+	Parameters []string
+}
+
+// RunCommandClient is a client for Azure Run Commands.
+type RunCommandClient interface {
```

Comment:
Prefer returning concrete types, let the users of your package define interfaces _if they need to_.

https://github.com/golang/go/wiki/CodeReviewComments#interfaces

Reply:
Bump this one? We could make the `runCommandClient` struct public and return it in place of the interface.

Reply:
Could we instead define the `RunCommandClient` interface in lib/cloud alongside `cloud.Clients`? It's tricky to mock the run command client if `cloud.Clients` forces implementers to use the concrete type.